api/container_delete.go

```diff
@@ -20,7 +20,7 @@ func (c *API) ContainerDelete(ctx context.Context, name string, force bool, dele
 
 	defer ignoreClose(res.Body)
 
-	if res.StatusCode == http.StatusNoContent {
+	if res.StatusCode == http.StatusNoContent || res.StatusCode == http.StatusOK {
```

Comment:
Is this fixing a bug you saw or is this somehow changed for this feature?

Reply:
I had to make this change to make a test pass. I'll have to check which one was failing to see if it was a bug. But it's not particularly related to the multiple podman socket feature.

Reply:
If I run `go test -v -race` without this extra check I get the following output/errors in all tasks:
```
2024-10-01T18:57:17.643+0100 [DEBUG] nomad-driver-podman/handle.go:79: podman.podman.default: Starting exitWatcher: container=328b6c260830fad1a9d87a05e7e4f480ca80af95eef71f33a0f600860a39d809
2024-10-01T18:57:17.643+0100 [DEBUG] nomad-driver-podman/handle.go:216: podman.podman.default: Container is not running anymore: container=328b6c260830fad1a9d87a05e7e4f480ca80af95eef71f33a0f600860a39d809 error="No such Container"
2024-10-01T18:57:18.644+0100 [DEBUG] nomad-driver-podman/handle.go:82: podman.podman.default: Stopping exitWatcher: container=328b6c260830fad1a9d87a05e7e4f480ca80af95eef71f33a0f600860a39d809
2024-10-01T18:57:18.724+0100 [WARN]  nomad-driver-podman/driver.go:1272: podman: Could not remove container: container=328b6c260830fad1a9d87a05e7e4f480ca80af95eef71f33a0f600860a39d809 error="cannot delete container, status code: 200"
```
It states "cannot delete container, status code: 200". The output from Podman is the following JSON:
```
[{"Id":"328b6c260830fad1a9d87a05e7e4f480ca80af95eef71f33a0f600860a39d809","RawInput":"328b6c260830fad1a9d87a05e7e4f480ca80af95eef71f33a0f600860a39d809"}]
```
Adding the extra check makes sure that if Podman gives a "200 OK" back, that we don't log this as an error.